api/src/OpenTelemetry/Internal/Logging/Types.hs

```diff
@@ -118,9 +154,8 @@ data ImmutableLogRecord body = ImmutableLogRecord
   -- ^ Additional information about the specific event occurrence. Unlike the Resource field, which is fixed for a particular source, Attributes can vary for each occurrence of the event coming from the same source.
   -- Can contain information about the request context (other than Trace Context Fields). The log attribute model MUST support any type, a superset of standard Attribute, to preserve the semantics of structured attributes
   -- emitted by the applications. This field is optional.
-  , logRecordLogger :: Logger
+  , logRecordLogger :: Logger body
   }
-  deriving (Functor)
```

Comment:
Functor is not used as far as I know and it is more complicated with the addition of `body` to `Logger`.

Reply:
Would be nice to have, but we can circle back to it.

Reply:
This was fixed with refactoring to LogRecord.